Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/site/SiteUrlHandler.java

```diff
@@ -16,21 +16,23 @@
  */
 package com.github.adamantcheese.chan.core.site;
 
-import androidx.annotation.Nullable;
-
-import com.github.adamantcheese.chan.core.model.Post;
 import com.github.adamantcheese.chan.core.model.orm.Loadable;
 
 import okhttp3.HttpUrl;
 
-public interface SiteUrlHandler {
-    Class<? extends Site> getSiteClass();
+public abstract class SiteUrlHandler {
+    abstract public Class<? extends Site> getSiteClass();
+
+    abstract public boolean matchesName(String value);
 
-    boolean matchesName(String value);
+    abstract public boolean respondsTo(HttpUrl url);
 
-    boolean respondsTo(HttpUrl url);
+    public String desktopUrlForThread(Loadable loadable) {
```

Comment:
I don't see a reason for having two methods here and changing the architecture so much. If you need the thread url, just set the post number to be the loadable number and have the various implementations deal with it. The only one that actually uses that postNo field is 4chan anyways.

Reply:
Well yes, but I think it's more obvious. If you want to share a thread you do need to pass a post.